### PR TITLE
Change workshop setup to use central Python virtual env

### PR DESCRIPTION
**Pull request by ofuhrer**
Use a central venv for all of the workshop participants allows us to silently and instantly push changes and new gt4py versions to all participants.

### CODE REVIEW

File: setup/create_venv [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+module purge
+source ${dir}/module_env
+
+# create virtual environment for workshop
+python -m venv venv
+source venv/bin/activate
+
+# install some standard packages we need
+pip install --upgrade pip setuptools wheel
+pip install numpy matplotlib ipykernel ipyparallel mpi4py
```
Comment (rheacangeo) [inline setup/create_venv:23]: hopefully in the next week the latest version of these don't change in ways that break... to be suuppper safe we might want to pin them

File: setup/create_venv [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# install GT4Py
+if [ ! -d gt4py ] ; then
+  #git clone git@github.com:VulcanClimateModeling/gt4py.git
+  git clone git@github.com:GridTools/gt4py.git
```
Comment (rheacangeo) [inline setup/create_venv:28]: hopefully we can switch this back before the sessions

File: setup/setup_venv [hunk 3]
```diff
@@ -1,23 +1,31 @@
 #!/bin/bash
 
+set -e
+
+venv_location=/home/ofuhrer/install/venv
+
+# get directory where this script lives
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```
Comment (rheacangeo) [inline setup/setup_venv:8]: bash magic!

**Review by rheacangeo — APPROVED** [review]
Looks good! I haven't tried it yet so don't know if I'll run into issues, but this look very useful!

File: setup/setup_venv (hunk 3 above)
Comment (ofuhrer) [reply setup/setup_venv:8]: With warm greetings from stackoverflow...

File: setup/create_venv (hunk 2 above)
Comment (ofuhrer) [reply setup/create_venv:28]: Agreed.

File: setup/create_venv (hunk 1 above)
Comment (ofuhrer) [reply setup/create_venv:23]: Good point. Probably having a requirements.txt file with pinned versions might be a good thing. Follow-up PR.

**Comment by ofuhrer** [comment]
Thank you @rheacangeo. I'll merge and be happy to help fix stuff if problems arise. It worked for me (I tested it), but it might cause issues because of access writes to my venv directory.

**Pull request merged**